ksqldb-engine/src/main/java/io/confluent/ksql/internal/ScalablePushQueryExecutorMetrics.java

```diff
@@ -0,0 +1,728 @@
+import org.apache.kafka.common.metrics.stats.Avg;
+import org.apache.kafka.common.metrics.stats.CumulativeCount;
+import org.apache.kafka.common.metrics.stats.CumulativeSum;
+import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.metrics.stats.Min;
+import org.apache.kafka.common.metrics.stats.Percentile;
+import org.apache.kafka.common.metrics.stats.Percentiles;
+import org.apache.kafka.common.metrics.stats.Percentiles.BucketSizing;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.utils.Time;
+
+@SuppressWarnings("checkstyle:ClassDataAbstractionCoupling")
+public class ScalablePushQueryExecutorMetrics implements Closeable {
+
+  private static final String SCALABLE_PUSH_QUERY_METRIC_GROUP = "scalable-push-query";
+  private static final String SCALABLE_PUSH_REQUESTS = "scalable-push-query-requests";
+  private static final long MAX_LATENCY_BUCKET_VALUE_MICROS = TimeUnit.SECONDS.toMicros(10);
+  private static final int NUM_LATENCY_BUCKETS = 1000;
+
+  private final List<Sensor> sensors;
+  private final Sensor localRequestsSensor;
+  private final Sensor remoteRequestsSensor;
+  private final Sensor connectionDurationSensor;
+  private final Map<MetricsKey, Sensor> connectionDurationSensorMap;
+  private final Sensor errorRateSensor;
+  private final Map<MetricsKey, Sensor> errorRateSensorMap;
+  private final Sensor requestSizeSensor;
+  private final Sensor responseSizeSensor;
+  private final Map<MetricsKey, Sensor> responseSizeSensorMap;
+  private final Sensor responseCode2XX;
+  private final Sensor responseCode3XX;
+  private final Sensor responseCode4XX;
+  private final Sensor responseCode5XX;
+  private final Map<MetricsKey, Sensor> rowsReturnedSensorMap;
+  private final Map<MetricsKey, Sensor> rowsProcessedSensorMap;
+  private final Metrics metrics;
+  private final Map<String, String> legacyCustomMetricsTags;
+  private final Map<String, String> customMetricsTags;
+  private final String ksqlServiceIdLegacyPrefix;
+  private final String ksqlServicePrefix;
+  private final Time time;
+
+  public ScalablePushQueryExecutorMetrics(
+      final String ksqlServiceId,
+      final Map<String, String> customMetricsTags,
+      final Time time
+  ) {
+    this.ksqlServiceIdLegacyPrefix = ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX
+        + ksqlServiceId;
+    this.legacyCustomMetricsTags =
+        Objects.requireNonNull(customMetricsTags, "customMetricsTags");
+
+    this.ksqlServicePrefix = ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX;
+    final Map<String, String> metricsTags = new HashMap<>(customMetricsTags);
+    metricsTags.put(KsqlConstants.KSQL_SERVICE_ID_METRICS_TAG, ksqlServiceId);
+    this.customMetricsTags = ImmutableMap.copyOf(metricsTags);
+
+    this.time = Objects.requireNonNull(time, "time");
+    this.metrics = MetricCollectors.getMetrics();
+    this.sensors = new ArrayList<>();
+    this.localRequestsSensor = configureLocalRequestsSensor();
```

Comment:
I don't think we need `localRequestsSensor` and `remoteRequestsSensor` if we're going to have the tag from RoutingNodeType.  These were legacy metrics for pull queries before we added these tags.